bibliopixel/remote/control.py

```diff
@@ -0,0 +1,140 @@
+
+        server_args = (
+            self.config['external_access'],
+            self.config['port'],
+            self.q_send,
+            self.q_recv
+        )
+
+        self.server = multiprocessing.Process(target=server.run_server,
+                                              args=server_args)
+
+        if not isinstance(animations, list):
+            raise ValueError('animations must be a list!')
+
+        self.current_animation_name = None
+        self.current_animation_obj = None
+        self.animations = animations
+        self.animation_objs = {}
+        anim_list = []
+        for anim in self.animations:
+            anim_cfg = dict(DEFAULT_ANIM_CONFIG)
+            anim_cfg.update(anim)
+            if not anim_cfg['display']:
+                anim_cfg['display'] = anim['name']
+            anim_cfg['name'] = ''.join(e for e in anim['name'] if e.isalnum())
+            anim_cfg['animation'].complete_callback = self.complete_callback
+            self.animation_objs[anim_cfg['name']] = anim_cfg['animation']
+            del anim_cfg['animation']  # no longer need here
+            anim_list.append(anim_cfg)
+        self.animations = anim_list
+
+        self.handlers = {}
+        # TODO: More pythonic way to do this?
+        for name, method in inspect.getmembers(self, predicate=inspect.ismethod):
+            if name.startswith('handler_'):
+                self.handlers[name.replace('handler_', '')] = method
+
+        self.default = base_config['default']
+        if issubclass(type(self.default), BaseAnimation):
+            self.animation_objs[DEFAULT_OFF] = self.default
+            self.default = DEFAULT_OFF
+
+        if self.default not in self.animation_objs:
+            raise ValueError(('`{}` is not a valid default! '
+                              'It must be one of the configured animation names.').format(self.default))
+
+    def cleanup(self):
+        self.q_recv.close()
+        self.q_send.close()
+        self.server.terminate()
+
+    def complete_callback(self, reason):
+        if reason != COMPLETE_REASON.CANCELED:
+            self.__start_default()
+
+    def __start_default(self):
+        if self.default:
+            self.__run_anim(self.default)
+
+    def __stop_anim(self, run_default=False):
+        if self.current_animation_obj:
+            self.current_animation_obj.cleanup(clean_layout=False)
+        if run_default:
+            self.__start_default()
+
+    def __start_anim(self, name):
+        self.__stop_anim()
+        self.__run_anim(name)
+
+    def __run_anim(self, name):
+        log.info('Running animation: {}'.format(name))
+        self.current_animation_name = name
+        self.current_animation_obj = self.animation_objs[name]
+        self.current_animation_obj.start()
+
+    def handler_run_animation(self, name):
+        if name not in self.animation_objs:
+            return False, 'Invalid animation name: {}'.format(name)
```

Comment:
It looks like handlers either return `False, 'error message'` on failure, or `True, None` on success.

Why not have them just return `'error message'` or `None`?